src/components/contact/components/contact-form/ContactForm.tsx

```diff
@@ -0,0 +1,131 @@
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useForm } from 'react-hook-form';
+import { useMutation } from '@tanstack/react-query';
+import { type ContactSchema, contactSchema } from '@/schemas/contact.schema';
+
+export const ContactForm: React.FC = () => {
+  const {
+    handleSubmit,
+    register,
+    setError,
+    formState: { errors },
+    reset,
+  } = useForm<ContactSchema>({
+    resolver: zodResolver(contactSchema),
+  });
+
+  const submitFormData = async (
+    data: ContactSchema,
+  ): Promise<{ message: string }> => {
+    const response = await fetch('/api/contact', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(data),
+    });
+
+    if (!response.ok) {
+      const errorData = await response.json();
+
+      throw new Error(JSON.stringify(errorData));
+    }
+
+    return response.json();
+  };
```

Comment:
think about this method as fetcher. Define this in `src/api/contact/module.ts`

Reply:
moved the fetcher to the contact/module.ts